### PR TITLE
Support for Solidity 0.8.22

### PR DESCRIPTION
**Pull request by blitz-1306**
## Changes
- [x] Updated dependencies.
- [x]  Added support of Solidity 0.8.22 on compiler level.
- [x] Tweaked affected components due to introduced feature of file-level `EventDefinitions` (`SourceUnit`, `InferType`, resolving, AST writing and file-level definitions parser).
- [x] Updated test snapshots.

## Notes
I did not discovered any breaking changes when updating Scribble (see https://github.com/Consensys/scribble/pull/251), so we can release a **patch version** of the package soon.

## Related links
- [Solidity 0.8.22 release notes](https://github.com/ethereum/solidity/releases/tag/v0.8.22)
- [Solidity 0.8.22 release blog post](https://soliditylang.org/blog/2023/10/25/solidity-0.8.22-release-announcement)

Regards.

### CODE REVIEW

File: src/compile/inference/file_level_definitions.pegjs [hunk 1]
```diff
@@ -143,6 +144,18 @@ EnumDef = ENUM __ name: Identifier __ body: EnumDefBody {
     } as FLEnumDefinition;
 }
 
+// ==== Event
+
+EventArgs = LPAREN __ ParenSoup __ RPAREN { return text(); }
+EventDef = EVENT __ name: Identifier __ args: EventArgs __ SEMICOLON {
```
Comment (cd1m0) [inline src/compile/inference/file_level_definitions.pegjs:150]: Small nit: The event definition can optionally have an `anonymous` keyword after the arguments. (See [here](https://docs.soliditylang.org/en/latest/grammar.html#a4.SolidityParser.eventDefinition) for details)

File: src/compile/inference/file_level_definitions_parser_header.ts [hunk 2]
```diff
@@ -65,6 +66,11 @@ export interface FLEnumDefinition extends FileLevelNode<FileLevelNodeKind.Enum>
     body: string;
 }
 
+export interface FLEventDefinition extends FileLevelNode<FileLevelNodeKind.Event> {
+    name: string;
+    args: string;
```
Comment (cd1m0) [inline src/compile/inference/file_level_definitions_parser_header.ts:71]: Should have a field `anonymous: bool`

File: test/samples/solidity/latest_08.sol [hunk 3]
```diff
@@ -385,3 +385,24 @@ contract Features_0819 {
...
+
+event X(uint a);
+
+contract Features_0822 {
+    event X(uint a);
+
+    function main() public {
+        emit IntEvents.X(1);
+        emit LibEvents.X(2);
+        emit X(3);
```
Comment (cd1m0) [inline test/samples/solidity/latest_08.sol:405]: This is a nit, not neccessary to fix. But I think this emit refers to the event defined in the contract, not the file-level event definition. So there is no emit that actually tests the file-level event def

**Review by cd1m0 — CHANGES_REQUESTED** [review]
Good work! Small nits to fix

File: test/samples/solidity/latest_08.sol (hunk 3 above)
Comment (blitz-1306) [reply test/samples/solidity/latest_08.sol:410]: Addressed in 8fdc8af

File: src/compile/inference/file_level_definitions_parser_header.ts (hunk 2 above)
Comment (blitz-1306) [reply src/compile/inference/file_level_definitions_parser_header.ts:71]: Addressed in 8fdc8af

File: src/compile/inference/file_level_definitions.pegjs (hunk 1 above)
Comment (blitz-1306) [reply src/compile/inference/file_level_definitions.pegjs]: Addressed in 8fdc8af

**Review by cd1m0 — APPROVED** [review]
LGTM

**Pull request merged**